web/overlay_manager.js

```diff
@@ -25,131 +25,143 @@
+        function OverlayManager_register(name, element, callerCloseMethod,
+                                         canForceClose) {
+      return new Promise(function (resolve) {
+        var container;
+        if (!name || !element || !(container = element.parentNode)) {
+          throw new Error('Not enough parameters.');
+        } else if (this.overlays[name]) {
+          throw new Error('The overlay is already registered.');
+        }
+        this.overlays[name] = { element: element,
+                                container: container,
+                                callerCloseMethod: (callerCloseMethod || null),
+                                canForceClose: (canForceClose || false) };
+        resolve();
+      }.bind(this));
+    },
 
-  /**
-   * @param {string} name The name of the overlay that should be opened.
-   * @returns {Promise} A promise that is resolved when the overlay has been
-   *                    opened.
-   */
-  open: function overlayManagerOpen(name) {
-    return new Promise(function (resolve) {
-      if (!this.overlays[name]) {
-        throw new Error('The overlay does not exist.');
-      } else if (this.active) {
-        if (this.overlays[name].canForceClose) {
-          this._closeThroughCaller();
+    /**
+     * @param {string} name The name of the overlay that is unregistered.
+     * @returns {Promise} A promise that is resolved when the overlay has been
+     *                    unregistered.
+     */
+    unregister: function OverlayManager_unregister(name) {
+      return new Promise(function (resolve) {
+        if (!this.overlays[name]) {
+          throw new Error('The overlay does not exist.');
         } else if (this.active === name) {
-          throw new Error('The overlay is already active.');
-        } else {
+          throw new Error('The overlay cannot be removed while it is active.');
+        }
+        delete this.overlays[name];
+
+        resolve();
+      }.bind(this));
+    },
+
+    /**
+     * @param {string} name The name of the overlay that should be opened.
+     * @returns {Promise} A promise that is resolved when the overlay has been
+     *                    opened.
+     */
+    open: function OverlayManager_open(name) {
+      return new Promise(function (resolve) {
+        if (!this.overlays[name]) {
+          throw new Error('The overlay does not exist.');
+        } else if (this.active) {
+          if (this.overlays[name].canForceClose) {
+            this._closeThroughCaller();
+          } else if (this.active === name) {
+            throw new Error('The overlay is already active.');
+          } else {
+            throw new Error('Another overlay is currently active.');
+          }
+        }
+        this.active = name;
+        this.overlays[this.active].element.classList.remove('hidden');
+        this.overlays[this.active].container.classList.remove('hidden');
+
+        window.addEventListener('keydown', this._keyDown.bind(this));
```

Comment:
Unfortunately you'll not be able to remove this event listener if you define it like this.
You'll need to add `this._keyDownBound = this._keyDown.bind(this);` to e.g. the constructor and then use `this._keyDownBound` here (and below).